### PR TITLE
fixes #3467 feat(nimbus): set up edit overview form, fill it with existing data

### LINKED ISSUE #3467: `PageEditOverview` form + display saved data
A dummy form was built out in EXP-296 and global data setup handled in EXP-297.

1) Add `FormOverviewPartial` or whatever component was created in EXP-296 to display on `PageEditOverview`
2) Display the fetched data from EXP-302 in the appropriate form fields

Other portions of this page will be done in EXP-481, we just need to display the same form here for now.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-298)
┆Issue Number: EXP-298

### LINKED ISSUE #3741: `PageEditOverview` form field modifications
EXP-296 made a `FormExperimentOverviewPartial` that we're displaying on `PageNew` and `PageOverview`.

On `PageEditOverview`, we need to:
1) Hide the "Public name" field (which can probably be done with an `if` check on if the slug exists)
2) Add the "Public Description" field

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-481)
┆Issue Number: EXP-481

### PR DESCRIPTION
**Pull request by jodyheavener**
This PR sets up the Experiment Overview form to work with the Edit Overview page, adds a public description field, and fills it with existing experiment data.

Closes #3467
Closes #3741

### CODE REVIEW

File: app/experimenter/nimbus-ui/src/components/FormExperimentOverviewPartial/index.test.tsx [hunk 1]
```diff
@@ -56,6 +78,18 @@ describe("FormExperimentOverviewPartial", () => {
...
 
+  const checkExistingForm = async (expected: Record<string, string>) => {
+    for (const [labelText, fieldValue] of [
+      ["Public name", expected.name],
+      ["Hypothesis", expected.hypothesis],
+      ["Application", expected.application],
+      ["Public Description", expected.publicDescription],
+    ]) {
+      const fieldName = screen.getByLabelText(labelText);
+      expect((fieldName as HTMLInputElement).value).toEqual(fieldValue);
```
Comment (jodyheavener) [inline app/experimenter/nimbus-ui/src/components/FormExperimentOverviewPartial/index.test.tsx:89]: @LZoog this is where the failure occurs, with the "Application" field, triggered by the "with an existing experiment..." test.

File: app/experimenter/nimbus-ui/src/components/FormExperimentOverviewPartial/index.tsx [hunk 2]
```diff
@@ -117,28 +151,53 @@ const FormExperimentOverviewPartial = ({
...
+          />
+          <Form.Text className="text-muted">
+            This description will be public to users on about:studies
+          </Form.Text>
+          <FormErrors name="publicDescription" />
+        </Form.Group>
+      )}
+
       <div className="d-flex flex-row-reverse bd-highlight">
+        {onNext && (
```
Comment (LZoog) [inline app/experimenter/nimbus-ui/src/components/FormExperimentOverviewPartial/index.tsx:171]: Do we want to display the Next button and the overview submit button on the Overview page outside of `OverviewPartial`? We're going to need to add one additional field post-parity (public description) and if we add that in this file, it's no longer a `partial` but instead is just a `FormOverview` displayed on the "overview" page and the "New" page with conditionals ("overview" shows a new field with a next/submit button, and the "new" page hides that field and instead shows a "create" button). 

If that makes things easier to not deal with nested components/callbacks with this being a "partial" of a form, maybe we should rename this file. Prior to yesterday the "public name" was also going to be hidden on the overview page so the separation between new/overview was slightly more significant.

Comment (LZoog) [reply app/experimenter/nimbus-ui/src/components/FormExperimentOverviewPartial/index.tsx:171]: Actually, after staring at this a bit longer I see the GQL call for creating an experiment does indeed live in `PageNew` and the query for `PageEditOverview` can live there so yeah this seems fine - but still wondering if the name of the component makes sense. (and I'm the one that originally suggested the name hehe)

File: app/experimenter/nimbus-ui/src/components/FormExperimentOverviewPartial/index.tsx [hunk 3]
```diff
@@ -117,28 +151,53 @@ const FormExperimentOverviewPartial = ({
...
+            as="textarea"
+            rows={3}
+            defaultValue={experiment?.publicDescription || ""}
+          />
+          <Form.Text className="text-muted">
+            This description will be public to users on about:studies
+          </Form.Text>
+          <FormErrors name="publicDescription" />
+        </Form.Group>
+      )}
```
Comment (LZoog) [inline app/experimenter/nimbus-ui/src/components/FormExperimentOverviewPartial/index.tsx:168]: Yep you also did #3741 😂

File: app/experimenter/nimbus-ui/src/components/FormOverview/index.test.tsx [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+  });
+
+  it("renders as expected", async () => {
+    render(<Subject />);
+    await act(async () => {
+      expect(screen.getByTestId("FormOverview")).toBeInTheDocument();
+    });
+  });
+
+  it("calls onCancel when cancel clicked", async () => {
```
Comment (LZoog) [inline app/experimenter/nimbus-ui/src/components/FormOverview/index.test.tsx:29]: Oh hey, we're also almost done with EXP-462 now

File: app/experimenter/nimbus-ui/src/components/HeaderEditExperiment/index.test.tsx [hunk 5]
```diff
@@ -28,4 +28,17 @@ describe("HeaderEditExperiment", () => {
       "DRAFT",
     );
   });
+  it("renders with no name when no name", () => {
```
Comment (LZoog) [inline app/experimenter/nimbus-ui/src/components/HeaderEditExperiment/index.test.tsx:31]: Is it possible for users to save an experiment without a name or is this leftover from removing the update-name-on-user-input functionality which can now be removed?

**Review by LZoog — APPROVED** [review]
LGTM!

File: app/experimenter/nimbus-ui/src/components/FormOverview/index.test.tsx (hunk 4 above)
Comment (jodyheavener) [reply app/experimenter/nimbus-ui/src/components/FormOverview/index.test.tsx:29]: Oooh I am so tempted to finish it, but it's not in the epic so I will resist for now

**Pull request merged**